test/zipkin/util.go

```diff
@@ -0,0 +1,135 @@
+/*
+Copyright 2018 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+//zipkin_util has constants and helper methods useful for zipkin tracing support.
+
+package zipkin
+
+import (
+	"errors"
+	"fmt"
+	"net"
+	"os"
+	"os/exec"
+	"strconv"
+
+	"github.com/knative/pkg/test/logging"
+	"go.opencensus.io/trace"
+	"k8s.io/client-go/kubernetes"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+const (
+
+	// ZipkinPortForwardPid environment variable name to store Process ID of the process started for zipkin port-foward.
+	ZipkinPortForwardPid = "ZIPKIN_PORTFWD_PID"
+
+	//ZipkinTraceIDHeader HTTP response header key to be used to store Zipkin Trace ID.
+	ZipkinTraceIDHeader = "ZIPKIN_TRACE_ID"
+
+	// ZipkinPort port exposed by the Zipkin Pod
+	ZipkinPort = 9411
+
+	// ZipkinTraceEndpoint port-forwarded zipkin endpoint
+	ZipkinTraceEndpoint = "http://localhost:9411/api/v2/trace/%s"
+
+	// ZipkinNamespace namespace where zipkin pod runs.
+	ZipkinNamespace = "istio-system"
+)
+
+// SetupZipkinTracing sets up zipkin tracing which involves a) Setting up port-forwarding from localhost to zipkin pod on the cluster (pid of the process doing Port-Forward is stored in an env variable).
+// b) enable AlwaysSample config for tracing.
+func SetupZipkinTracing(kubeClientset *kubernetes.Clientset) error {
```

Comment:
In the design doc we discussed about having the zipkin traces on by default. Your PR makes it optional. What was the conclusion?

Reply:
As discussed in the design doc, zipkin traces would be enabled by default. Only thing is that following this change I would be updating the Setup and TearDown methods in test/e2e.go. The reason for this is the port-forward process required for Zipkin traces. There is no clean way in SpoofingClient to clean up this process. Adding it in the client would mean each test has to be changed, hence having it the test/e2e.go would be cleaner way to enable zipkin tracing by default and also clean up after test completion.

Reply:
SGTM.